### ISSUE TITLE
pypi: Datasource unknown error - Cannot read properties of undefined (reading 'replace')

### ISSUE THREAD

**Issue by morremeyer** [mend/renovate-on-prem#298]
With version 5.0.0, we're getting the following error for pypi datasource lookups (and only those):

```plain
DEBUG: Datasource unknown error (repository=our-org/the-repo)
       "datasource": "pypi",
       "packageName": "boto3",
       "err": {
         "message": "Cannot read properties of undefined (reading 'replace')",
         "stack": "TypeError: Cannot read properties of undefined (reading 'replace')\n    at PypiDatasource.getReleases (/usr/src/app/node_modules/renovate/dist/modules/datasource/pypi/index.js:28:68)\n    at fetchReleases (/usr/src/app/node_modules/renovate/dist/modules/datasource/index.js:260:36)\n    at getRawReleases (/usr/src/app/node_modules/renovate/dist/modules/datasource/index.js:288:25)\n    at getPkgReleases (/usr/src/app/node_modules/renovate/dist/modules/datasource/index.js:304:40)\n    at getPkgReleasesWithResult (/usr/src/app/node_modules/renovate/dist/modules/datasource/index.js:377:33)\n    at lookupUpdates (/usr/src/app/node_modules/renovate/dist/workers/repository/process/lookup/index.js:51:83)\n    at /usr/src/app/node_modules/renovate/dist/workers/repository/process/fetch.js:62:115\n    at withLookupStats (/usr/src/app/node_modules/renovate/dist/workers/repository/process/fetch.js:19:26)\n    at fetchDepUpdates (/usr/src/app/node_modules/renovate/dist/workers/repository/process/fetch.js:62:44)\n    at async /usr/src/app/node_modules/p-map/index.js:57:22"
       }
```

This is the following line in the source https://github.com/renovatebot/renovate/blob/a4ac402e3c281e41d0f531a1398b31ee91b9cb7b/lib/modules/datasource/pypi/index.ts#L39.

I cannot reproduce this on versions 36.0.0 (which I think is the renovate version for on-prem version 5.0.0) or 36.8.5 (the most recent version as of writing) when running against the same repository locally VSCode (following https://github.com/renovatebot/renovate/blob/a4ac402e3c281e41d0f531a1398b31ee91b9cb7b/docs/development/local-development.md#vs-code). 

Is it possible that this is a problem with only the on-prem version?

**Comment by rarkins** [comment]
I can't think of any reason why Renovate On-Prem would have this version while regular Renovate would not.

Renovate On-Prem v5.0.0 is using renovate@36.1.11. If the problem doesn't reproduce on that, the best thing would be to add a defensive check and log message to Renovate and then publish a new On-Prem release once that's ready.

**Comment by morremeyer** [comment]
I found the issue. We're using the `existingSecret` configuration since we create the secret outside of the on-prem deployment.

Therefore, we need to set the the `pipIndexUrl` since it is required by https://github.com/mend/renovate-on-prem/blob/ea3b60031e9dbd40e3e280b5002d4b5f626af075/helm-charts/whitesource-renovate/templates/deployment.yaml#L97-L103. 

But, since we do not have any custom config here, we just set an empty string. This sets an empty environment variable, effectively breaking the datasource.

I'm working on a PR, will open that soon.
